engine/src/main/java/org/destinationsol/components/SurvivesWithZeroHealth.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020 The Terasology Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.components;
+
+import org.terasology.gestalt.entitysystem.component.Component;
+
+/**
+ * If an entity with a {@link Health} component that has zero health should not be destroyed, it should have
+ * this component.
+ */
+public class SurvivesWithZeroHealth implements Component<SurvivesWithZeroHealth> {
```

Comment:
I don't particularly like this. Generally, IMO, entities should have features based on what components they have. Here, however, you are basing feature on the _lack_ of component. Not only is this a bit ugly, it is also quite error-prone - it can be quite easy to forget about the existence of particular component.